### DISCUSSION TITLE
Teslamate in Portainer / how to upgrade

### DISCUSSION THREAD

**Discussion by dansche** [teslamate-org/teslamate#1671] [Q&A] [answered]
Hi community

First of all I love Teslamate and lots of my Teslamates have it installed because of me.

I'm not a Raspi prof., can do alot with Youtube help ;-) But this is mostly like cut and paste... Anyway I have a portainer running with some services and Teslamate is one of them. 

Can someone tell me what's the best way to update now Teslamate to the most actual version without losing the old data, I tried with https://docs.teslamate.org/docs/upgrading but I was not able to do a backup with that https://docs.teslamate.org/docs/maintenance/backup_restore/ I tried to login with Putty to run the backup command but I receive an error "root@rasp4portainer:/home/pi/Teslamate# docker-compose exec -T database pg_dump -U teslamate teslamate > /backuplocation/teslamate.bck
bash: /backuplocation/teslamate.bck: No such file or directory"

I can't imagine, that I am the only one using portainer so if there is someone experienced I would love to hear from you if possible step-by-step how to upgrade.

Thanks alot

**Comment by DrMichael** [comment, marked as answer]
> `docker-compose exec -T database pg_dump -U teslamate teslamate > /backuplocation/teslamate.bck`
> bash: /backuplocation/teslamate.bck: No such file or directory

Right, because /backuplocation does not exist. Just use `docker-compose exec -T database pg_dump -U teslamate teslamate > teslamate.bck`

> I can't imagine, that I am the only one using portainer so if there is someone experienced I would love to hear from you if possible step-by-step how to upgrade.

That all has nothing to do with portainer, I am also running it in the teslamate stack...

Michael

**Reply by dansche** [reply]
Cool, that did the trick went smoothly and all the data is still there!!!

Great, Thank you Michael

For all other falling over that threat:
If you have it up an running in portainer, just enter with Putty the Raspberry

Enter the Teslamate directory: cd Teslamate
Copy: docker-compose exec -T database pg_dump -U teslamate teslamate > teslamate.bck
This makes the data backup, I copied it away with FileZilla to my local harddrive just something happens in case....

Copy: docker-compose pull
Wait until all is green and shows "done"

Copy: docker-compose up -d
Wait till everything is done and you are all set

On my side all the data was still there and I am on the most acutal release.

Again thanks Michael for your help

**Reply by DrMichael** [reply]
You're welcome. You might want to close this here then.

**Comment by zikronix** [comment]
Im trying to just move the database to a new pi. This backup doesnt work for me at all, I get a zero byte file. Ive tried running it from the pi directly in terminal, or via ssh. Even connected to the instance in portainer. any help would be great

**Reply by DrMichael** [reply]
What gives
`docker-compose exec -T database pg_dump -U teslamate teslamate`?

**Reply by zikronix** [reply]
I was able to get it to back up using sudo docker exec <dbname> pg_dump -U teslamate teslamate > backup.bck

However now can’t get it to restore I was able to dump the tables and schema replacing docker-compose with docker exec. But trying restoring doesn’t give me any output. I did verify that I do have compose installed.
